packages/sync/src/firehose/index.ts

```diff
@@ -0,0 +1,342 @@
+import { createDeferrable, Deferrable, wait } from '@atproto/common'
+import {
+  IdResolver,
+  parseToAtprotoDocument,
+  DidDocument,
+} from '@atproto/identity'
+import {
+  cborToLexRecord,
+  formatDataKey,
+  parseDataKey,
+  readCar,
+  RepoVerificationError,
+  verifyProofs,
+} from '@atproto/repo'
+import { AtUri } from '@atproto/syntax'
+import { Subscription } from '@atproto/xrpc-server'
+import {
+  type Account,
+  type Commit,
+  type Identity,
+  type RepoEvent,
+  RepoOp,
+  isAccount,
+  isCommit,
+  isIdentity,
+  isValidRepoEvent,
+} from './lexicons'
+import {
+  Event,
+  CommitMeta,
+  CommitEvt,
+  AccountEvt,
+  AccountStatus,
+  IdentityEvt,
+} from '../events'
+import { CID } from 'multiformats/cid'
+import { SyncQueue } from '../queue'
+import { didAndSeqForEvt } from '../util'
+
+export type FirehoseOptions = {
+  idResolver: IdResolver
+
+  handleEvt: (evt: Event) => Promise<void>
```

Comment:
Just a suggestion, but I'd lean towards spelling this out in the public API.
```suggestion
  handleEvent: (evt: Event) => Promise<void>
```

Reply:
yea you right